Description: partial backport of r260554 (79fe19caf)
 In particular, we need -DU_SHOW_CPLUSPLUS_API=0 to avoid compilation
 errors in C++ parts of ICU headers which we are not using anyway.
Origin: upstream, https://github.com/qtwebkit/qtwebkit/commit/756e1c8f23dc2720
Last-Update: 2024-05-28

--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -1137,6 +1137,13 @@
 #define ENABLE_PLATFORM_FONT_LOOKUP 1
 #endif
 
+/* FIXME: This does not belong in Platform.h and should instead be included in another mechanism (compiler option, prefix header, config.h, etc) */
+/* ICU configuration. Some of these match ICU defaults on some platforms, but we would like them consistently set everywhere we build WebKit. */
+#define U_SHOW_CPLUSPLUS_API 0
+#ifdef __cplusplus
+#define UCHAR_TYPE char16_t
+#endif
+
 #if COMPILER(MSVC)
 #undef __STDC_LIMIT_MACROS
 #define __STDC_LIMIT_MACROS
